File: common/src/test/java/com/hartwig/actin/treatment/input/composite/CompositeRulesTest.java
package com.hartwig.actin.treatment.input.composite;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import com.hartwig.actin.treatment.datamodel.EligibilityRule;

import org.junit.Test;

public class CompositeRulesTest {

    @Test
    public void canInterpretCompositeRules() {
        EligibilityRule firstCompositeRule = CompositeRules.COMPOSITE_RULE_MAPPING.keySet().iterator().next();
        assertTrue(CompositeRules.isComposite(firstCompositeRule));
        assertNotNull(CompositeRules.inputsForCompositeRule(firstCompositeRule));

        assertFalse(CompositeRules.isComposite(EligibilityRule.HAS_ACTIVE_INFECTION));
    }
}

File: common/src/test/java/com/hartwig/actin/treatment/input/composite/CompositeRulesTest.kt
package com.hartwig.actin.treatment.input.composite

import com.hartwig.actin.treatment.datamodel.EligibilityRule
import com.hartwig.actin.treatment.input.composite.CompositeRules.inputsForCompositeRule
import com.hartwig.actin.treatment.input.composite.CompositeRules.isComposite
import org.junit.Assert
import org.junit.Test

class CompositeRulesTest {
    @Test
    fun canInterpretCompositeRules() {
        val firstCompositeRule = CompositeRules.COMPOSITE_RULE_MAPPING.keys.iterator().next()
        Assert.assertTrue(isComposite(firstCompositeRule))
        Assert.assertNotNull(inputsForCompositeRule(firstCompositeRule))
        Assert.assertFalse(isComposite(EligibilityRule.HAS_ACTIVE_INFECTION))
    }
}